deltacat/aws/s3u.py

```diff
@@ -335,8 +343,12 @@ def upload_table(
         except ClientError as e:
             if e.response["Error"]["Code"] == "NoSuchKey":
                 # s3fs may swallow S3 errors - we were probably throttled
-                raise RetryableError(f"Retry table upload to: {s3_url}") from e
-            raise NonRetryableError(f"Failed table upload to: {s3_url}") from e
+                raise RetryableError(
```

Comment:
You can throw a specific exception here that subclasses `RetryableError`. Technically, this is also a `UploadTableError`.